### PR TITLE
feat(integrations): add hubspot integration

### PR DESCRIPTION
**Pull request by pashafateev**

### CODE REVIEW

File: web/static/embed.go [hunk 1]
```diff
@@ -52,6 +52,9 @@ var GoogleSheetsWebContent embed.FS
 //go:embed i/http/connect
 var HTTPWebContent embed.FS
 
+//go:embed hubspot/connect
+var HubspotWebContent embed.FS
```
Comment (daabr) [inline web/static/embed.go]:
```suggestion
var HubSpotWebContent embed.FS
```

File: runtimes/pythonrt/py-sdk/autokitteh/hubspot.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"""Initialize a Hubspot client, based on an AutoKitteh connection."""
```
Comment (daabr) [inline runtimes/pythonrt/py-sdk/autokitteh/hubspot.py]:
```suggestion
"""Initialize a HubSpot client, based on an AutoKitteh connection."""
```

File: runtimes/pythonrt/py-sdk/autokitteh/hubspot.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import os
+
+from hubspot import HubSpot
+
+from .connections import check_connection_name, refresh_oauth
+from .errors import ConnectionInitError, OAuthRefreshError
+
+
+def hubspot_client(connection: str, **kwargs) -> HubSpot:
+    """Initialize a Hubspot client, based on an AutoKitteh connection.
```
Comment (daabr) [inline runtimes/pythonrt/py-sdk/autokitteh/hubspot.py]:
```suggestion
    """Initialize a HubSpot client, based on an AutoKitteh connection.
```

File: runtimes/pythonrt/py-sdk/autokitteh/hubspot.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    Args:
+        connection: AutoKitteh connection name.
+
+    Returns:
+        Hubspot SDK client.
```
Comment (daabr) [inline runtimes/pythonrt/py-sdk/autokitteh/hubspot.py]:
```suggestion
        HubSpot SDK client.
```

File: runtimes/pythonrt/py-sdk/autokitteh/hubspot.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+    Raises:
+        ValueError: AutoKitteh connection name is invalid.
+        ConnectionInitError: AutoKitteh connection was not initialized yet.
+    """
+    check_connection_name(connection)
+
+    access_token = os.getenv(connection + "__oauth_AccessToken")
```
Comment (daabr) [inline runtimes/pythonrt/py-sdk/autokitteh/hubspot.py]: This `access_token` isn't actually being used. Instead, you should check if the refresh token exists (`connection + "__oauth_RefreshToken"`).

File: runtimes/pythonrt/py-sdk/autokitteh/hubspot.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    Args:
+        connection: AutoKitteh connection name.
+
+    Returns:
+        Hubspot SDK client.
+
+    Raises:
+        ValueError: AutoKitteh connection name is invalid.
+        ConnectionInitError: AutoKitteh connection was not initialized yet.
```
Comment (daabr) [inline runtimes/pythonrt/py-sdk/autokitteh/hubspot.py:22]:
```suggestion
        ConnectionInitError: AutoKitteh connection was not initialized yet.
        OAuthRefreshError: OAuth token refresh failed.
```

File: runtimes/pythonrt/py-sdk/autokitteh/hubspot.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    check_connection_name(connection)
+
+    access_token = os.getenv(connection + "__oauth_AccessToken")
+    if not access_token:
+        raise ConnectionInitError("OAuth access token is missing")
+
+    try:
+        access_token, _ = refresh_oauth("hubspot", connection)
+    except Exception as e:
+        raise OAuthRefreshError(connection, str(e)) from e
```
Comment (daabr) [inline runtimes/pythonrt/py-sdk/autokitteh/hubspot.py:34]: `raise from` - nice! I was not aware of this option. Please add it in other relevant places in the SDK (in a separate PR).

File: internal/backend/oauth/oauth.go [hunk 8]
```diff
@@ -285,6 +285,25 @@ func New(l *zap.Logger) sdkservices.OAuth {
 				},
 			},
 
+			"hubspot": {
+				ClientID:     os.Getenv("HUBSPOT_CLIENT_ID"),
+				ClientSecret: os.Getenv("HUBSPOT_CLIENT_SECRET"),
```
Comment (daabr) [inline internal/backend/oauth/oauth.go:292]: Is there a ticket to add these vars to staging and prod? (And store their management link and secrets in 1Password)

File: internal/backend/oauth/oauth.go [hunk 9]
```diff
@@ -285,6 +285,25 @@ func New(l *zap.Logger) sdkservices.OAuth {
 				},
 			},
 
+			"hubspot": {
```
Comment (daabr) [inline internal/backend/oauth/oauth.go:290]:
```suggestion
			// Based on:
			// https://developers.hubspot.com/beta-docs/guides/apps/authentication/working-with-oauth
			"hubspot": {
```

File: integrations/hubspot/server.go [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+	"go.autokitteh.dev/autokitteh/web/static"
+)
+
+const (
+	// oauthPath is the URL path for our handler to save new OAuth-based
+	// connections, after users submit them via a web form.
+	oauthPath = "/hubspot/oauth"
+)
+
+// Start initializes all the HTTP handlers of the Hubspot integration.
```
Comment (daabr) [inline integrations/hubspot/server.go]:
```suggestion
// Start initializes all the HTTP handlers of the HubSpot integration.
```

File: integrations/hubspot/client.go [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+	"go.autokitteh.dev/autokitteh/sdk/sdkintegrations"
+	"go.autokitteh.dev/autokitteh/sdk/sdkmodule"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+type integration struct{ vars sdkservices.Vars }
+
+var (
+	authType      = sdktypes.NewSymbol("authType")
```
Comment (daabr) [inline integrations/hubspot/client.go]: Apparently, the web UI sets and expects `auth_type` and not `authType` - so we have both in many integrations. No need to keep this error in new integrations.

```suggestion
	authType      = sdktypes.NewSymbol("auth_type")
```

**Review by daabr — APPROVED** [review]
Approved with various cosmetic comments/suggestions, and a few general questions.

File: internal/backend/oauth/oauth.go (hunk 8 above)
Comment (pashafateev) [reply internal/backend/oauth/oauth.go:292]: Who do I open the ticket for?

File: runtimes/pythonrt/py-sdk/autokitteh/hubspot.py (hunk 7 above)
Comment (pashafateev) [reply runtimes/pythonrt/py-sdk/autokitteh/hubspot.py:34]: https://linear.app/autokitteh/issue/INT-115/add-raise-from-to-relevant-places-in-python-sdk

File: runtimes/pythonrt/py-sdk/autokitteh/hubspot.py (hunk 5 above)
Comment (pashafateev) [reply runtimes/pythonrt/py-sdk/autokitteh/hubspot.py]: Done

File: internal/backend/oauth/oauth.go (hunk 8 above)
Comment (daabr) [reply internal/backend/oauth/oauth.go:292]: See INT-8 for example

**Pull request merged**